cmd/oras/internal/display/metadata/interface.go

```diff
@@ -20,17 +20,17 @@ import (
 	"oras.land/oras/cmd/oras/internal/option"
 )
 
+// Renderer renders metadata information when an operation is complete.
+type Renderer interface {
+	Render() error
+}
+
 // PushHandler handles metadata output for push events.
 type PushHandler interface {
 	TaggedHandler
 
-	OnCopied(opts *option.Target) error
-	OnCompleted(root ocispec.Descriptor) error
-}
-
-// Renderer renders metadata information when an operation is complete.
-type Renderer interface {
-	Render() error
+	OnCopied(opts *option.Target, root ocispec.Descriptor) error
+	Renderer
```

Comment:
Suggest to move `Render up` with `TaggedHandler`

Reply:
I'll do it in the last PR of this issue, otherwise there will be too many file changes as `TaggedHandler` is used in many metadata handlers.